src/cio_file.c

```diff
@@ -1049,6 +1049,13 @@ int cio_file_fs_size_change(struct cio_file *cf, size_t new_size)
          * fallocate() is not portable, Linux only.
          */
         ret = fallocate(cf->fd, 0, 0, new_size);
+        if (ret == -1) {
+            /* If fallocate fails try operation using posix_fallocate. Required
+            * since some filesystems do not support the fallocate operation e.g.
+            * ext3 and reiserfs.
+            */
+            ret = posix_fallocate(cf->fd, 0, new_size);
```

Comment:
from fallocate(2) manpage (on error):

```
       EOPNOTSUPP
              The filesystem containing the file referred to by fd does not support this operation; or the mode is not supported by the filesystem containing the file referred to by fd.
```

so the new approach should check for ```EOPNOTSUPP``` otherwise it would assume that any failure, including when the disk is full, it's because a missing interface in the VFS layer.

Reply:
That would make sense... even though the ENOSPC would be returned by the posix_fallocate method called.

One practical thing. What is your recommended best practice after me applying the suggested fix. Shall I rebase the commits to one or not?

Reply:
we don't need to perform two system calls, they are expensive! :) 

You can squash it

Reply:
Its address now as per review.